File: 19-07-2025-Remove_Sub-Folders_from_the_Filesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {

        sort(folder.begin(),folder.end());
         
        vector<string>res;

        for(string &path : folder)
        {
            if(res.size()==0 || path.find(res.back()+"/") !=0)
            {
                res.push_back(path);
            }
        }
        

        return res;
    }
};

File: 19-07-2025_Remove_Sub-Folders_from_the_Filesystem.java
class Solution {
    public List<String> removeSubfolders(String[] folder) 
    { 

          Arrays.sort(folder); 

        List<String> res = new ArrayList<>();

        for (String path : folder) {
            if (res.isEmpty() || !path.startsWith(res.get(res.size() - 1) + "/")) {
                res.add(path);
            }
        }

        return res;

    }
}
